x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java

```diff
@@ -16,24 +18,40 @@
 import org.elasticsearch.xpack.security.transport.filter.SecurityIpFilterRule;
 
 import java.net.InetAddress;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.Collections;
 import java.util.List;
 
 public class AuditTrailService {
 
+    private static final Logger logger = LogManager.getLogger(AuditTrailService.class);
+
     private static final AuditTrail NOOP_AUDIT_TRAIL = new NoopAuditTrail();
     private final CompositeAuditTrail compositeAuditTrail;
     private final XPackLicenseState licenseState;
+    private final Duration minLogPeriod = Duration.ofMinutes(30);
+    // protected for tests
+    protected Instant lastLogInstant = Instant.EPOCH;
 
     public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState licenseState) {
         this.compositeAuditTrail = new CompositeAuditTrail(Collections.unmodifiableList(auditTrails));
         this.licenseState = licenseState;
     }
 
     public AuditTrail get() {
-        if (compositeAuditTrail.isEmpty() == false &&
-            licenseState.isSecurityEnabled() && licenseState.checkFeature(Feature.SECURITY_AUDITING)) {
-            return compositeAuditTrail;
+        if (compositeAuditTrail.isEmpty() == false) {
+            if (licenseState.isSecurityEnabled() && licenseState.checkFeature(Feature.SECURITY_AUDITING)) {
```

Comment:
Technically, the license level check is just `licenseState.checkFeature`, so I'd prefer to have the logic as
```
if (compositeAuditTrail.isEmpty() == false && licenseState.isSecurityEnabled()) {
    if (licenseState.checkFeature(Feature.SECURITY_AUDITING)) {
    }
}
```
which is more accurate to the warning message.